app/models/city.rb

```diff
@@ -23,28 +18,45 @@ def has_valid_data
     return false
   end
       
-      
   def update_city_data
-    location_key = self.location_key
-    if self.updated_at <= Date.today.to_time.beginning_of_day or !has_valid_data()
-      url = "http://dataservice.accuweather.com/forecasts/v1/daily/5day/#{location_key}"
+    if self.updated_at.nil? or self.updated_at <= Date.today.to_time.beginning_of_day or !has_valid_data()
+      url = "http://dataservice.accuweather.com/forecasts/v1/daily/5day/#{self.location_key}"
       query = {apikey: City.get_accuweather_key(), language:"en-us", details: "true"}
-      response = City.get_resonse(HTTParty.get(url, query: query), url, query)
-      self.update_attribute("daily_data" , response)
+      resp = HTTParty.get(url, query: query)
+      self.update_attribute("daily_data" , resp)
     end
   end
   
+  def self.obtain_loc_key(lat, lng)
+    # We want to avoid creating a City object when we get location key
+    url = "http://dataservice.accuweather.com/locations/v1/cities/geoposition/search"
+    query = {apikey: City.get_accuweather_key(), q: "#{lat},#{lng}",language:"en-us" }
+    resp = HTTParty.get(url, query: query)
+    return resp["Key"]
+  end
+  
   def self.get_loc_key(lat,lng, name)
     city = City.find_by(lat: lat, lng: lng)
     if city and !city.location_key.nil?
       return city.location_key
     end
     url = "http://dataservice.accuweather.com/locations/v1/cities/geoposition/search"
     query = {apikey: City.get_accuweather_key(), q: "#{lat},#{lng}",language:"en-us" }
-    response = City.get_resonse(HTTParty.get(url, query: query), url, query)
-    location_key = response["Key"]
+    resp = HTTParty.get(url, query: query)
+    location_key = resp["Key"]
```

Comment:
Not DRY: replace lines 43-46 with `self.obtain_loc_key`

Reply:
noted